src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -197,6 +196,7 @@ object TaskBuilder {
     // args take precedence over command, if supplied
     app.args.foreach { argv =>
       builder.setShell(false)
+      builder.setValue(argv.head)
```

Comment:
It did not work for the Mesos containerizer, but probably for the docker containerizer. It looks like the protobuf fields (cmd, args) have a different format with a different containerizer. We could deal with that easily, but we use the protobufs also for persisting the AppDefinition (and that needs to be synchronous (write/read)).